### PR TITLE
WIP: GPD Win Max 2 support

### PR DESCRIPTION
**Pull request by ajwahab**
Runs and exits cleanly now.  Tweaked gyro toggle.  A few formatting and stylistic modifications, mainly for conciseness and clarity.  Took some steps to remove subprocess calls since those can be dangerous.  Still need to find a means of testing gyro functionality.

### CODE REVIEW

File: handycon.py [hunk 1]
```diff
@@ -613,12 +619,7 @@ async def capture_keyboard_events():
                             elif active == [] and seed_event.code in [10] and button_on == 0 and button2 in event_queue:
                                 this_button = button2
                                 await do_rumble(0, 150, 1000, 0)
-
```
Comment (pastaq) [inline handycon.py:616]: Why was this removed? There is currently a bug in Steam that broke the OSK shortcut, but I'd prefer to operate under the assumption that it will be fixed.

File: handycon.py [hunk 2]
```diff
@@ -746,15 +747,16 @@ async def capture_power_events():
                     active_keys = keyboard_device.active_keys()
                     if event.type == e.EV_KEY and event.code == 116: # KEY_POWER
                         if event.value == 0:
+                            steam_path = HOME_PATH / '.steam/root/ubuntu12_32/steam'
                             if active_keys == [125]:
                                 # For DeckUI Sessions
                                 shutdown = True
-                                cmd = 'su {} -c "{}/.steam/root/ubuntu12_32/steam -ifrunning steam://longpowerpress"'.format(USER, HOME_PATH)
+                                cmd = f'su {USER} -c "{steam_path} -ifrunning steam://longpowerpress"'
```
Comment (pastaq) [inline handycon.py:754]: Did you test this? short press should show an animation and long press should pop up the power menu.

File: handycon.py [hunk 3]
```diff
@@ -7,18 +7,19 @@
...
-import logging
-import os
-import signal
-import subprocess
-import warnings
 import configparser
-
 from evdev import InputDevice, InputEvent, UInput, ecodes as e, list_devices, ff
-from pathlib import PurePath as p
+import logging
```
Comment (pastaq) [inline handycon.py:12]: I prefer to have all the full imports at the top, then partial libarary imports, then local imports. This helps readability a lot when you have lots of imports.

**Comment by pastaq** [comment]
>  Still need to find a means of testing gyro functionality.
the `evtest` package should work. when enabled you should get right stick ABS events at all times, vice only when moving right stick otherwise.

File: handycon.py (hunk 2 above)
Comment (ajwahab) [reply handycon.py:754]: Short press put my machine to sleep (expected desktop environment behavior).  Long press forced shutdown (expected BIOS behavior).

File: handycon.py (hunk 1 above)
Comment (ajwahab) [reply handycon.py:616]: I removed it since the WM2 has a physical keyboard and [11] was being used to toggle the gyro.

File: handycon.py (hunk 2 above)
Comment (pastaq) [reply handycon.py:754]: Was there an animation for short press? I have a catch for regular suspend if we're in desktop mode, but steam does some magic that makes resuming a game in progress more stable.


"Long press" is about 2 seconds

File: handycon.py (hunk 1 above)
Comment (pastaq) [reply handycon.py:616]: Oh, right. that makes sense. We can map that to another event instead. Screenshot might be a good one, or we could simplify the logic for gyro to that event only.

**Comment by ajwahab** [comment]
> > Still need to find a means of testing gyro functionality.
> 
> the `evtest` package should work. when enabled you should get right stick ABS events at all times, vice only when moving right stick otherwise.

This appears to work as you've described.

File: handycon.py [hunk 4]
```diff
@@ -55,19 +56,20 @@
...
 cmd = "who | awk '{print $1}' | sort | head -1"
-while USER == None:
-    USER_LIST = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout = subprocess.PIPE, stderr = subprocess.PIPE, shell=True)
-    for get_first in USER_LIST.stdout:
-        name = get_first.decode().strip()
-        if name is not None:
-            USER = name
-        break
+while USER is None:
+    who = [w.split() for w in os.popen('who').read().strip().split('\n')]
```
Comment (pastaq) [inline handycon.py:66]: I receive the following error when running this

```
Traceback (most recent call last):
  File "/home/gamer/forks/HandyGCCS/./handycon.py", line 67, in <module>
    who.sort(key=lambda row: (row[2], row[3]))
  File "/home/gamer/forks/HandyGCCS/./handycon.py", line 67, in <lambda>
    who.sort(key=lambda row: (row[2], row[3]))
IndexError: list index out of range
```

File: handycon.py [hunk 5]
```diff
@@ -120,7 +122,7 @@ def id_system():
 
     # Identify the current device type. Kill script if not compatible.
     system_id = open("/sys/devices/virtual/dmi/id/product_name", "r").read().strip()
-    system_cpu = subprocess.check_output("lscpu | grep \"Vendor ID\" | cut -d : -f 2 | xargs", shell=True, universal_newlines=True).strip()
+    system_cpu = platform.processor()
```
Comment (pastaq) [inline handycon.py:125]: We aren't using this anymore actually, we can get rid of it and the platform import.

**Review by pastaq — CHANGES_REQUESTED** [review]

File: handycon.py (hunk 4 above)
Comment (ajwahab) [reply handycon.py:66]:
```python
>>> import os
>>> who = [w.split() for w in os.popen('who').read().strip().split('\n')]
>>> who
[['awahab', 'pts/1', '2022-11-23', '17:56', '(IP REDACTED)']]
[['awahab', 'pts/0', '2022-11-23', '17:53', '(IP REDACTED)']]
>>> who.sort(key=lambda row: (row[2], row[3]))  # prioritize sort by date (row[2]) then time (row[3])
>>> who
[['awahab', 'pts/0', '2022-11-23', '17:53', '(IP REDACTED)']]
[['awahab', 'pts/1', '2022-11-23', '17:56', '(IP REDACTED)']]
>>> who[0][0]  # pick username of lowest index
'awahab'
```

Comment (pastaq) [reply handycon.py:66]: it appears there is inconsistency between your OS and ChimeraOS here

```
>>> import os
>>> who = [w.split() for w in os.popen('who').read().strip().split('\n')]
>>> who
[['gamer', '2022-11-23', '14:38'], ['gamer', 'pts/0', '2022-11-23', '14:41', '(REDACTED)']]
>>> who.sort(key=lambda row: (row[2], row[3]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <lambda>
IndexError: list index out of range
>>> 
```

Comment (ajwahab) [reply handycon.py:66]: That's bizarre.  I tested with Python 3.8 and 3.10 on Ubuntu and Arch, respectively.  What version of Python 3 is Chimera using?
How about `who = sorted(who, key=lambda row: (row[2], row[3]))` ?

Comment (pastaq) [reply handycon.py:66]: Its not a python issue. `'who` in bash produces the bad output.

```
[gamer@chimeraos ~]$ who
gamer                 2022-11-23 14:15
gamer    pts/0        2022-11-23 16:21 (IP.ADD.RES.S)
```

Comment (pastaq) [reply handycon.py:66]: I believe the bottom entry is my SSH session and the top is the default session.

Comment (ajwahab) [reply handycon.py]: Of course!  Sorry, was reading on a small screen and missed a column or two.  Just pushed a revised approach.

**Review by pastaq — APPROVED** [review]
